### PR TITLE
Added a large number of external dependencies

### PR DESCRIPTION
**Pull request by Vermorkentech**
Added a large number of external dependencies used by axon framework and extensions to prevent mismatches with incompatible versions.

This is a follow-up of adding the reactor dependency to the bom in https://github.com/AxonFramework/axon-bom/pull/43

I have added a (very) large number of additional dependencies, which are all used by the Axon Framework and/or the various extensions. However, there a couple of points for discussion:

1. To what extent do we want to add these other dependecies? I have added just about everything I could see as a candidate for now, but believe that, especially for framework, we may not want to include all of these dependencies (see comment in pom.xml for the most dubious dependencies IMO) and limit this to those that are most likely to end up with version mismatches and related issues.
2. Instead of limiting this list, we may want to consider offering two versions of the bom as an alternative, one with just the base Axon dependencies (`axon-bom`), and another with all the secondary dependencies included as well (`axon-bom-extended` or something similar). Personally I don't have any strong preference here.

Based on the conclusions we might draw form the above points, we'll have to revisit/cleanup this pom a bit as well, especially versions through variables vs plain versions using the `<version>` tag without referencing a variable.

### CODE REVIEW

**Comment by azzazzel** [comment]
I would advise against this. It is not clear to me what problem it solves. I looked at both #43 and #11, but I couldn't find the actual root issue. 

If developers only use the framework (and the extensions), they don't need to add those dependencies. If they do add them, it's because something else needs them. That something likely has its expectations about the correct versions. 

The purpose of the BOM is to make things easier. It does not enforce anything nor prevent from doing something. Developers can still provide a dependency version explicitly in their POMs. Furthermore, they could add another BOM with a different version of the same dependency without realizing it. And that is precisely why I advise against this approach. We shouldn't make people wonder where this version came from if it is an artifact that we haven't released ourselves.

If the issue is that the framework or a module is not compatible with a specific version of a particular dependency, we should make that clear in a place that is easy to find. Ultimately, the end-user is responsible for solving the dependency conflict in whatever way works best for them. The BOM itself will not help anyone who is not aware of why there is an issue in the first place.

If we want to help developers discover and deal with such known issues, we can provide instructions on installing and configuring [Maven Enforcer Plugin](https://maven.apache.org/enforcer/maven-enforcer-plugin/index.html). It has a built-in [Dependency Convergence rule](https://maven.apache.org/enforcer/enforcer-rules/dependencyConvergence.html) to detect such problems. Or we can [provide our own](https://maven.apache.org/enforcer/enforcer-api/writing-a-custom-rule.html) rule.

**Comment by Vermorkentech** [comment]
https://github.com/AxonFramework/axon-bom/issues/11 originated from a problem one of our more advanced users experienced with version mismatches which caused them some issues. I wasn't involved in those original discussions though, so I don't know the exact details leading to the issue. However, from what I've gathered later, it is my understanding that this was indeed an issue with the framework not being compatible with a different reactor verion pulled in by on of our own extensions or vice versa (which has been fixed by now). @MGathier You might know a little more on this, so please correct me if I'm wrong in my last statement.

Since the above got fixed an should really be a one-time issue, I'd also be fine leaving these adittions out completely (or adding it to a different bom to keep this one clean with releases we manage only as suggested in my point 2). However, in that case I do think it will be worthwhile to investigate your last points, give some more guidance to our users and especially be more explicit in some way if there are incompatible versions that the framework or one of the extensions cannot deal with. Perhaps some section in the reference guide listing known incompatible versions would also help here.

I'm fine either way, but one thing I do feel strongly about is that we should be consistent in this, whatever we choose. If we decide not to add any of these dependencies, then we should also revert https://github.com/AxonFramework/axon-bom/pull/43

**Review by gklijs — APPROVED** [review]
Looks good to me. versions can always be overwritten, but this way it's easier for users of the framework to stay up to date + they have some guarantee the versions used here will work.

**Comment by smcvb** [comment]
Have any of us investigated what other providers of BOM's do? 

I'm not overly biased, similarly to @Vermorkentech, as long as we're clear and consistent.
I do think tagging along with other BOM releases would be beneficial, since that enforces an expectation BOM-users would have.

If none provide additional dependencies of artifacts they don't own, than we take the route suggested by @azzazzel.
If several do provide dependencies of used artifacts, then I don't see why we shouldn't.

File: pom.xml [hunk 1]
```diff
@@ -139,6 +173,92 @@
...
+                <groupId>io.dropwizard.metrics</groupId>
+                <artifactId>metrics-core</artifactId>
+                <version>${dropwizard.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.micrometer</groupId>
+                <artifactId>micrometer-core</artifactId>
+                <version>${micrometer.version}</version>
+            </dependency>
+            <!-- Do we want these in here as well? -->
```
Comment (smcvb) [inline pom.xml:211]: I would say no. :-)

File: pom.xml [hunk 2]
```diff
@@ -259,6 +433,77 @@
...
+                <groupId>io.opentracing.contrib</groupId>
+                <artifactId>opentracing-spring-tracer-configuration-starter</artifactId>
+                <version>${opentracing-spring-tracer.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>io.opentracing.contrib</groupId>
+                <artifactId>opentracing-spring-jaeger-web-starter</artifactId>
+                <version>${opentracing-spring-jaeger.version}</version>
+            </dependency>
+            <!-- Spring Dependencies -->
```
Comment (smcvb) [inline pom.xml:446]: I'd wager we wouldn't want these in here either.

**Comment by smcvb** [comment]
After some internal debate, we, for now, think it's best not to pursue this idea further.
As such, I will close this (draft) pull request.

Regardless, feel free to keep commenting here if you have an opinion on the matter.

**Comment by gklijs** [comment]
Since the, I certainly changed my opinion. Even the Reactor one could be doubted, since the latest does break when Spring security is used with reactor. And although they can always be overriden directly, it can sometimes be hard to know wherer a certain dependency is comming from. Moreover I feel like we should have some guarantee that all the version in the pom privided by us work well together. which is hard to oversee apart from the artifacts we realaese ourselves.

**Pull request closed without merging**